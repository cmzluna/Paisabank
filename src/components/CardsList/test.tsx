import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import CardsListComponent from ".";
import MastercardIcon from "assets/icons/issuers/mastercard.svg";
import VisaIcon from "assets/icons/issuers/visa.svg";

const mockedData = [
  {
    balance: "978,85",
    currency: "USD",
    expDate: "2026-03-20",
    id: 1,
    issuer: "mastercard",
    lastDigits: 1234,
    name: "Soy Paisanx",
    svgFile: MastercardIcon,
  },
  {
    balance: "1000,10",
    currency: "USD",
    expDate: "2027-03-20",
    id: 2,
    issuer: "visa",
    lastDigits: 1234,
    name: "Soy Paisanx",
    svgFile: VisaIcon,
  },
];

const mockProps = { data: mockedData };

describe("returns error", () => {
  it("returns null when data prop is not passed", () => {
    const ActionsTabComponent = renderer.create(<CardsListComponent />).toJSON();
    expect(ActionsTabComponent).toBeNull();
  });
});

describe("renders correctly", () => {
  it("match snapshot", () => {
    const ActionsTabComponent = renderer
      .create(<CardsListComponent data={mockedData} isLoading={false} />)
      .toJSON();
    expect(ActionsTabComponent).toMatchSnapshot();
  });

  it("renders loading indicator if loading prop is true", () => {});

  it("renders all items correctly", () => {
    const { getAllByTestId } = render(<CardsListComponent data={mockedData} isLoading={false} />);

    expect(getAllByTestId("CardItemImage")).toHaveLength(2);
  });
});
