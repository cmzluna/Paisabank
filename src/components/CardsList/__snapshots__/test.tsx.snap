// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly match snapshot 1`] = `
<View
  style={
    {
      "height": 282,
      "marginLeft": 43,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "balance": "978,85",
          "currency": "USD",
          "expDate": "2026-03-20",
          "id": 1,
          "issuer": "mastercard",
          "lastDigits": 1234,
          "name": "Soy Paisanx",
          "svgFile": "<svg width="35" height="25" viewBox="0 0 35 25" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="Logo">
<circle id="Ellipse 33" cx="12.2229" cy="12.2229" r="12.2229" fill="#E9231A"/>
<circle id="Ellipse 34" opacity="0.8" cx="21.7885" cy="12.2229" r="12.2229" fill="#E99418"/>
</g>
</svg>
",
        },
        {
          "balance": "1000,10",
          "currency": "USD",
          "expDate": "2027-03-20",
          "id": 2,
          "issuer": "visa",
          "lastDigits": 1234,
          "name": "Soy Paisanx",
          "svgFile": "<svg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="icons8-visa 1">
<path id="Vector" d="M45 35C45 37.209 43.209 39 41 39H7C4.791 39 3 37.209 3 35V13C3 10.791 4.791 9 7 9H41C43.209 9 45 10.791 45 13V35Z" fill="#2100C4"/>
<path id="Vector_2" d="M15.186 19L12.56 26.832C12.56 26.832 11.893 23.519 11.827 23.103C10.332 19.692 8.126 19.882 8.126 19.882L10.726 30V29.998H13.887L18.258 19H15.186ZM17.689 30H20.56L22.296 19H19.389L17.689 30ZM38.008 19H34.987L30.277 30H33.129L33.717 28.429H37.313L37.619 30H40.232L38.008 19ZM34.513 26.328L36.076 22.171L36.894 26.328H34.513ZM26.369 22.206C26.369 21.6 26.867 21.149 28.295 21.149C29.223 21.149 30.286 21.823 30.286 21.823L30.752 19.514C30.752 19.514 29.394 18.999 28.061 18.999C25.042 18.999 23.485 20.443 23.485 22.271C23.485 25.577 27.464 25.124 27.464 26.822C27.464 27.113 27.233 27.786 25.576 27.786C23.914 27.786 22.817 27.177 22.817 27.177L22.322 29.393C22.322 29.393 23.385 29.999 25.439 29.999C27.498 29.999 30.354 28.459 30.354 26.247C30.354 23.586 26.369 23.394 26.369 22.206Z" fill="white"/>
<path id="Vector_3" d="M12.212 24.945L11.246 20.197C11.246 20.197 10.809 19.168 9.673 19.168C8.537 19.168 5.233 19.168 5.233 19.168C5.233 19.168 10.894 20.84 12.212 24.945Z" fill="#F5BC00"/>
</g>
</svg>
",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#005CEE",
              "borderBottomLeftRadius": 14,
              "borderBottomRightRadius": 14,
              "borderTopLeftRadius": 14,
              "borderTopRightRadius": 14,
              "marginRight": 21,
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
              "width": 634,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 14,
                }
              }
            >
              Balance
            </Text>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="48"
              bbWidth="48"
              fill="none"
              focusable={false}
              height="48"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 48,
                    "width": 48,
                  },
                ]
              }
              testID="CardItemImage"
              vbHeight={25}
              vbWidth={35}
              width="48"
              xml="<svg width="35" height="25" viewBox="0 0 35 25" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="Logo">
<circle id="Ellipse 33" cx="12.2229" cy="12.2229" r="12.2229" fill="#E9231A"/>
<circle id="Ellipse 34" opacity="0.8" cx="21.7885" cy="12.2229" r="12.2229" fill="#E99418"/>
</g>
</svg>
"
              xmlns="http://www.w3.org/2000/svg"
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  name="Logo"
                >
                  <RNSVGCircle
                    cx="12.2229"
                    cy="12.2229"
                    fill={
                      {
                        "payload": 4293468954,
                        "type": 0,
                      }
                    }
                    name="Ellipse 33"
                    propList={
                      [
                        "fill",
                      ]
                    }
                    r="12.2229"
                  />
                  <RNSVGCircle
                    cx="21.7885"
                    cy="12.2229"
                    fill={
                      {
                        "payload": 4293497880,
                        "type": 0,
                      }
                    }
                    name="Ellipse 34"
                    opacity={0.8}
                    propList={
                      [
                        "fill",
                      ]
                    }
                    r="12.2229"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 11,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 11,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 30,
                  "justifyContent": "center",
                  "marginRight": 18,
                  "width": 48,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="30"
                bbWidth="48"
                fill="none"
                focusable={false}
                height={30}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 30,
                      "width": 48,
                    },
                  ]
                }
                vbHeight={30}
                vbWidth={48}
                width={48}
                xml="<svg width="48" height="30" viewBox="0 0 48 30" fill="none" xmlns="http://www.w3.org/2000/svg">
<path id="Rectangle 2 Copy 2" d="M40.8 0H7.2C3.22355 0 0 2.68629 0 6V24C0 27.3137 3.22355 30 7.2 30H40.8C44.7765 30 48 27.3137 48 24V6C48 2.68629 44.7765 0 40.8 0Z" fill="url(#paint0_linear_1_58)"/>
<defs>
<linearGradient id="paint0_linear_1_58" x1="26.5" y1="24.5" x2="15.7136" y2="1.39843" gradientUnits="userSpaceOnUse">
<stop offset="0.107247" stop-color="#89A5FB"/>
<stop offset="1" stop-color="#6C8FF8"/>
</linearGradient>
</defs>
</svg>
"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M40.8 0H7.2C3.22355 0 0 2.68629 0 6V24C0 27.3137 3.22355 30 7.2 30H40.8C44.7765 30 48 27.3137 48 24V6C48 2.68629 44.7765 0 40.8 0Z"
                    fill={
                      {
                        "brushRef": "paint0_linear_1_58",
                        "type": 1,
                      }
                    }
                    name="Rectangle 2 Copy 2"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGDefs>
                    <RNSVGLinearGradient
                      gradient={
                        [
                          0.107247,
                          -7756293,
                          1,
                          -9662472,
                        ]
                      }
                      gradientTransform={null}
                      gradientUnits={1}
                      name="paint0_linear_1_58"
                      x1="26.5"
                      x2="15.7136"
                      y1="24.5"
                      y2="1.39843"
                    />
                  </RNSVGDefs>
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "position": "absolute",
                  }
                }
              >
                USD
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 22,
                }
              }
            >
              978,85
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 11,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 11,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 22,
                }
              }
            >
              **** **** **** 
            </Text>
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 22,
                }
              }
            >
              1234
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16,
                  "height": 37,
                }
              }
            >
              Soy Paisanx
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                  }
                }
              >
                Exp. Date
              </Text>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 13,
                  }
                }
              >
                2026-03-20
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#005CEE",
              "borderBottomLeftRadius": 14,
              "borderBottomRightRadius": 14,
              "borderTopLeftRadius": 14,
              "borderTopRightRadius": 14,
              "marginRight": 21,
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
              "width": 634,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 14,
                }
              }
            >
              Balance
            </Text>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="48"
              bbWidth="48"
              fill="none"
              focusable={false}
              height="48"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 48,
                    "width": 48,
                  },
                ]
              }
              testID="CardItemImage"
              vbHeight={48}
              vbWidth={48}
              width="48"
              xml="<svg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="icons8-visa 1">
<path id="Vector" d="M45 35C45 37.209 43.209 39 41 39H7C4.791 39 3 37.209 3 35V13C3 10.791 4.791 9 7 9H41C43.209 9 45 10.791 45 13V35Z" fill="#2100C4"/>
<path id="Vector_2" d="M15.186 19L12.56 26.832C12.56 26.832 11.893 23.519 11.827 23.103C10.332 19.692 8.126 19.882 8.126 19.882L10.726 30V29.998H13.887L18.258 19H15.186ZM17.689 30H20.56L22.296 19H19.389L17.689 30ZM38.008 19H34.987L30.277 30H33.129L33.717 28.429H37.313L37.619 30H40.232L38.008 19ZM34.513 26.328L36.076 22.171L36.894 26.328H34.513ZM26.369 22.206C26.369 21.6 26.867 21.149 28.295 21.149C29.223 21.149 30.286 21.823 30.286 21.823L30.752 19.514C30.752 19.514 29.394 18.999 28.061 18.999C25.042 18.999 23.485 20.443 23.485 22.271C23.485 25.577 27.464 25.124 27.464 26.822C27.464 27.113 27.233 27.786 25.576 27.786C23.914 27.786 22.817 27.177 22.817 27.177L22.322 29.393C22.322 29.393 23.385 29.999 25.439 29.999C27.498 29.999 30.354 28.459 30.354 26.247C30.354 23.586 26.369 23.394 26.369 22.206Z" fill="white"/>
<path id="Vector_3" d="M12.212 24.945L11.246 20.197C11.246 20.197 10.809 19.168 9.673 19.168C8.537 19.168 5.233 19.168 5.233 19.168C5.233 19.168 10.894 20.84 12.212 24.945Z" fill="#F5BC00"/>
</g>
</svg>
"
              xmlns="http://www.w3.org/2000/svg"
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  name="icons8-visa 1"
                >
                  <RNSVGPath
                    d="M45 35C45 37.209 43.209 39 41 39H7C4.791 39 3 37.209 3 35V13C3 10.791 4.791 9 7 9H41C43.209 9 45 10.791 45 13V35Z"
                    fill={
                      {
                        "payload": 4280352964,
                        "type": 0,
                      }
                    }
                    name="Vector"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M15.186 19L12.56 26.832C12.56 26.832 11.893 23.519 11.827 23.103C10.332 19.692 8.126 19.882 8.126 19.882L10.726 30V29.998H13.887L18.258 19H15.186ZM17.689 30H20.56L22.296 19H19.389L17.689 30ZM38.008 19H34.987L30.277 30H33.129L33.717 28.429H37.313L37.619 30H40.232L38.008 19ZM34.513 26.328L36.076 22.171L36.894 26.328H34.513ZM26.369 22.206C26.369 21.6 26.867 21.149 28.295 21.149C29.223 21.149 30.286 21.823 30.286 21.823L30.752 19.514C30.752 19.514 29.394 18.999 28.061 18.999C25.042 18.999 23.485 20.443 23.485 22.271C23.485 25.577 27.464 25.124 27.464 26.822C27.464 27.113 27.233 27.786 25.576 27.786C23.914 27.786 22.817 27.177 22.817 27.177L22.322 29.393C22.322 29.393 23.385 29.999 25.439 29.999C27.498 29.999 30.354 28.459 30.354 26.247C30.354 23.586 26.369 23.394 26.369 22.206Z"
                    fill={
                      {
                        "payload": 4294967295,
                        "type": 0,
                      }
                    }
                    name="Vector_2"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M12.212 24.945L11.246 20.197C11.246 20.197 10.809 19.168 9.673 19.168C8.537 19.168 5.233 19.168 5.233 19.168C5.233 19.168 10.894 20.84 12.212 24.945Z"
                    fill={
                      {
                        "payload": 4294294528,
                        "type": 0,
                      }
                    }
                    name="Vector_3"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 11,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 11,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 30,
                  "justifyContent": "center",
                  "marginRight": 18,
                  "width": 48,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="30"
                bbWidth="48"
                fill="none"
                focusable={false}
                height={30}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 30,
                      "width": 48,
                    },
                  ]
                }
                vbHeight={30}
                vbWidth={48}
                width={48}
                xml="<svg width="48" height="30" viewBox="0 0 48 30" fill="none" xmlns="http://www.w3.org/2000/svg">
<path id="Rectangle 2 Copy 2" d="M40.8 0H7.2C3.22355 0 0 2.68629 0 6V24C0 27.3137 3.22355 30 7.2 30H40.8C44.7765 30 48 27.3137 48 24V6C48 2.68629 44.7765 0 40.8 0Z" fill="url(#paint0_linear_1_58)"/>
<defs>
<linearGradient id="paint0_linear_1_58" x1="26.5" y1="24.5" x2="15.7136" y2="1.39843" gradientUnits="userSpaceOnUse">
<stop offset="0.107247" stop-color="#89A5FB"/>
<stop offset="1" stop-color="#6C8FF8"/>
</linearGradient>
</defs>
</svg>
"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M40.8 0H7.2C3.22355 0 0 2.68629 0 6V24C0 27.3137 3.22355 30 7.2 30H40.8C44.7765 30 48 27.3137 48 24V6C48 2.68629 44.7765 0 40.8 0Z"
                    fill={
                      {
                        "brushRef": "paint0_linear_1_58",
                        "type": 1,
                      }
                    }
                    name="Rectangle 2 Copy 2"
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGDefs>
                    <RNSVGLinearGradient
                      gradient={
                        [
                          0.107247,
                          -7756293,
                          1,
                          -9662472,
                        ]
                      }
                      gradientTransform={null}
                      gradientUnits={1}
                      name="paint0_linear_1_58"
                      x1="26.5"
                      x2="15.7136"
                      y1="24.5"
                      y2="1.39843"
                    />
                  </RNSVGDefs>
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 14,
                    "position": "absolute",
                  }
                }
              >
                USD
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 22,
                }
              }
            >
              1000,10
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginBottom": 11,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 11,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 22,
                }
              }
            >
              **** **** **** 
            </Text>
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 22,
                }
              }
            >
              1234
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16,
                  "height": 37,
                }
              }
            >
              Soy Paisanx
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 10,
                  }
                }
              >
                Exp. Date
              </Text>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 13,
                  }
                }
              >
                2027-03-20
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
